### ISSUE TITLE
[Feature Request] Temporal Controler Panel

### ISSUE THREAD

**Issue by jakimowb** [EnMAP-Box/enmap-box#277]
QGIS uses the [QgsTemporalControllerWidget](https://api.qgis.org/api/classQgsTemporalControllerWidget.html) to manage the temporal properties of its central [QgsMapCanvas](https://api.qgis.org/api/classQgsMapCanvas.html) via:

![grafik](https://user-images.githubusercontent.com/1404870/204094905-261ece0b-d378-4627-99fb-607e31f60d80.png)

```
void QgsMapCanvas::setTemporalController(QgsTemporalController * controller) 	
void QgsMapCanvas::setTemporalRange(const QgsDateTimeRange & range) 	
```

Following the multi-mapview approach realized in the EnMAP-Box, it would make sense to have similar panel that:
* provides at minimum the same functions as known from the QgsTemporalControlerWidget, i.e. 
  * disabled/fixed/anminated temporal navigation
  * flexible specification of tempral ranges
* allows to link a temporal controler with _multiple_ map canvases. This aims to show spatial data from the same temporal range in different maps, i.e the same year but different spatial regions or spectral band combinations
* allows to define _multiple temporal controlers_. This way the same spatial regions can be compared showing different temporal ranges (example for can be found in the [EO Time Series Viewer code](https://bitbucket.org/jakimowb/eo-time-series-viewer) )

This may be implemented as an extendable list of QgsTemporalControlerWidgets and a matrix that allows to link these controller witth EnMAP-Box MapCanvases.

![Untitled-2022-11-26-1603](https://user-images.githubusercontent.com/1404870/204095754-e29d26bc-d583-4986-8698-468731f04532.png)

**Comment by janzandr** [comment]
I would suggest to only have a single Temporal Controller.

**Comment by janzandr** [comment]
Also, I would not temporally control at a MapView-level, but rather individual layer, like known from QGIS.
![image](https://user-images.githubusercontent.com/45030715/204302459-80bab366-9e80-4515-8fc4-0098e4953b76.png)

**Comment by janzandr** [comment]
Hi @jakimowb, before you start implementing this, please let's meet and discuss details.

**Comment by jakimowb** [comment]
I don't get your point. 
The temporal controler in QGIS controls "at MapView-level", i.e. the QgsMapCanvas filters all layers which are out-of-date range. 
That's what I like to see in the EnMAP-Box as well. The only difference is, that we should have a multi-map solution that allows to modifiy map date ranges individually.
In any case, the temporal properties of a data source are set at the map layer level (and as known from QGIS), e.g. using the layer properties dialog or specialized EnMAP-Box widgets / object. For example a renderer that is aware of the QgsMapCanvas temporal range and selects the bands automatically.

**Comment by janzandr** [comment]
> The temporal controler in QGIS controls "at MapView-level"

No, not true. Here you see 3 QGIS Map Views, all are controled by the single Temporal Controller panel:
![image](https://user-images.githubusercontent.com/45030715/204776396-3415c9e9-e8a0-443d-8ada-3ceb06003bc7.png)
 
I would suggest, that we stick to the same concept for the EnMAP-Box.
Otherwise it may get to complicated.

**Comment by jakimowb** [comment]
Now I understand what you mean. My feature request is to enhance this concept and allow to show different time frames in different maps canvases. Why/what do you think it is "to complicated"?

**Comment by janzandr** [comment]
> Why do you think it is "to complicated"?

Having more than one Temporal Controller is to complicated.
I would stick to the basic concept introduced by QGIS: all layers, that are under temporal control, are controlled by THE Temporal Controller, no matter in what MapView they are displayed.
